weave/trace/weave_init.py

```diff
@@ -107,13 +109,17 @@ def init_weave(
     wandb_api.init()
     wandb_context = wandb_api.get_wandb_api_context()
     if wandb_context is None:
-        import wandb
-
         logger.info(
             "Please login to Weights & Biases (https://wandb.ai/) to continue..."
         )
         wandb_termlog_patch.ensure_patched()
-        wandb.login(anonymous="never", force=True)  # type: ignore
+        try:
+            import wandb
+        except (ImportError, ModuleNotFoundError):
+            from weave import wandb_thin as wandb  # type: ignore[no-redef]
+        finally:
+            wandb.login(anonymous="never", force=True, host=host)  # type: ignore
```

Comment:
a few things:
1. Let's not do this try/catch here. Let's create a module (`wandb_thin` maybe) that exposes methods which do the switching under the hood. Let's not put it on the caller to do this dance.
2. adding `host` here implies that we will not pick up the `WANDB_BASE_URL` env var. I think that is a miss. We should have the thin version behave as identically to the real version as possible.

Reply:
I'd also like to see some tests for different hosts. there is a pretty complicated switch in `wandb` to handle the different host patterns